commons-test/src/main/java/org/infinispan/commons/test/RunningTestsRegistry.java

```diff
@@ -80,24 +93,60 @@ private static void killLongTest(Thread testThread, String testName, String simp
                ps.waitFor(10, SECONDS);
             }
 
-            File dumpFile =
-               new File(String.format("threaddump-%1$s-%2$tY-%2$tm-%2$td-%3$s.log", safeTestName, new Date(), pid));
-            System.out.printf("Dumping thread stacks of process %s to %s\n", pid, dumpFile.getAbsolutePath());
-            Process jstack = new ProcessBuilder()
-               .command(System.getProperty("java.home") + "/../bin/jstack", pid)
-               .redirectOutput(dumpFile)
-               .start();
-            jstack.waitFor(10, SECONDS);
-
             index++;
          }
+         return pids;
+      } catch (Exception e) {
+         System.err.println("Error collecting child processes:");
+         e.printStackTrace(System.err);
+         return Collections.emptyList();
+      }
+   }
+
+   private static void dumpThreads(String safeTestName, List<String> pids) {
+      try {
+         File jstackFile = new File(System.getProperty("java.home") + "/../bin/jstack");
+         if (!jstackFile.canExecute()) {
+            jstackFile = new File(System.getProperty("java.home") + "/bin/jstack");
+         }
+         LocalDateTime now = LocalDateTime.now();
+         if (jstackFile.canExecute() && !pids.isEmpty()) {
+            for (String pid : pids) {
+               File dumpFile = new File(String.format("threaddump-%1$s-%2$tY-%2$tm-%2$td-%3$s.log",
+                                                      safeTestName, now, pid));
+               System.out.printf("Dumping thread stacks of process %s to %s\n", pid, dumpFile.getAbsolutePath());
+               Process jstack = new ProcessBuilder()
+                                   .command(System.getProperty("java.home") + "/../bin/jstack", pid)
```

Comment:
Surely this should be jstackFile.getAbsolutePath()

Reply:
Yes, I was focusing on testing the custom thread dump code and missed this